### PR TITLE
CCD-4600: Upgrade to Java 17 and App insights v3

### PR DESCRIPTION
**Pull request by tomxelliott**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CCD-4600

### Change description ###
Upgrade to Java 17 and App insights v3

### CODE REVIEW

File: Jenkinsfile_CNP [hunk 1]
```diff
@@ -12,8 +12,8 @@ def branchesToSync = ['demo', 'ithc', 'perftest']
 GradleBuilder builder = new GradleBuilder(this, product)
 
 // Variables to switch pipeline logic and wiring per type of build
-def definitionStoreDevelopPr = "PR-575"  // This doesn't change frequently, but when it does, only change this value.
-def dataStoreApiDevelopPr    = "PR-1260" // This doesn't change frequently, but when it does, only change this value.
+def definitionStoreDevelopPr = "PR-1392"  // This doesn't change frequently, but when it does, only change this value.
+def dataStoreApiDevelopPr    = "PR-2267" // This doesn't change frequently, but when it does, only change this value.
```
Comment (RebeccaBaker) [inline Jenkinsfile_CNP:16]: to be reverted before merging

File: charts/ccd-case-document-am-api/values.preview.template.yaml [hunk 2]
```diff
@@ -2,4 +2,4 @@ java:
   image: ${IMAGE_NAME}
   ingressHost: ${SERVICE_FQDN}
   environment:
-    CCD_DATA_STORE_API_BASE_URL: http://ccd-data-store-api-pr-1260-java
+    CCD_DATA_STORE_API_BASE_URL: http://ccd-data-store-api-pr-2267-java
```
Comment (RebeccaBaker) [inline charts/ccd-case-document-am-api/values.preview.template.yaml:5]: to be reverted before merging

Comment (dinesh1patel) [reply charts/ccd-case-document-am-api/values.preview.template.yaml]: reverted

File: Jenkinsfile_CNP (hunk 1 above)
Comment (dinesh1patel) [reply Jenkinsfile_CNP]: reverted

**Review by dinesh1patel — APPROVED** [review]
LGTM

**Pull request merged**